pkg/route/api/types.go

```diff
@@ -38,6 +42,14 @@ type RouteSpec struct {
 	TLS *TLSConfig
 }
 
+// RouteTargetReference specifies the target that resolve into endpoints. Only the 'Service'
+// kind is allowed. Use 'weight' field to emphasize one over others.
+type RouteTargetReference struct {
+	Kind string
+	Name string
+	Weight int
```

Comment:
Yes, 1.3 will require us to have internal and external size-specific ints in order to enable protobuf.